src/lib/core/components/visualizations/implementations/BoxplotVisualization.tsx

```diff
@@ -145,6 +140,23 @@ function BoxplotViz(props: Props) {
 
   const findEntityAndVariable = useFindEntityAndVariable(entities);
 
+  const { xAxisVariable, yAxisVariable, overlayVariable } = useMemo(() => {
+    const xAxisVariable = findEntityAndVariable(vizConfig.xAxisVariable);
+    const yAxisVariable = findEntityAndVariable(vizConfig.yAxisVariable);
+    const overlayVariable = findEntityAndVariable(vizConfig.overlayVariable);
+
+    return {
+      xAxisVariable: xAxisVariable ? xAxisVariable.variable : undefined,
+      yAxisVariable: yAxisVariable ? yAxisVariable.variable : undefined,
+      overlayVariable: overlayVariable ? overlayVariable.variable : undefined,
+    };
+  }, [
+    findEntityAndVariable,
+    vizConfig.xAxisVariable,
+    vizConfig.yAxisVariable,
+    vizConfig.overlayVariable,
+  ]);
```

Comment:
@bobular I just wonder if this approach at boxplot viz will be used for all other vizs?

Reply:
Yes it might remove some repetitive use of `findEntityAndVariable` in other viz's - but boxplot had the most, so I only did it there so far.

Reply:
@bobular Thank you for your explanation. Since my axis label unit work is also related to this in part, perhaps I can do that for you if you want - the unit work needs to change all vizs as well :) I am waiting for this PR is merged so that I can avoid possible conflicts.